src/lib/deploy.js

```diff
@@ -243,8 +247,33 @@ async function deploy({ alias, yes }) {
   const verificationKey = await step(
     'Generate verification key (takes 10-30 sec)',
     async () => {
-      let { verificationKey } = await zkApp.compile(zkAppAddress);
-      return verificationKey;
+      let cache = fs.readJsonSync(`${DIR}/build/cache.json`);
+      // compute a hash of the contract's circuit to determine if 'zkapp.compile' should re-run
+      let currentDigest = await zkApp.digest(zkAppAddress);
+
+      // initialize cache if 'zk deploy' is run the first time on the contract
+      if (!cache[contractName]) {
+        cache[contractName] = { digest: '', verificationKey: '' };
+        fs.writeJsonSync(`${DIR}/build/cache.json`, cache, { spaces: 2 });
```

Comment:
Also, here you can skip writing the JSON, since if  `!cache[contractName]` then you will end up in the else branch below, where you compute the verification key and write back to disk anyway